### PR TITLE
Fix and update testing documentation

### PR DESCRIPTION
**Pull request by mattjmcnaughton**
Improve the testing documentation in the following ways:

1. Ensure all of the given commands actually work. Some of them
specified tests that no longer existed or packages that no longer had
tests.
2. Clarify the ability to run tests for a package and all its
subpackages with `/...`.
3. Clarify how the developer can also utilize `go test` if they desire.

### CODE REVIEW

**Comment by mattjmcnaughton** [comment]
/assign @stevekuznetsov 
thanks!

File: contributors/devel/sig-testing/testing.md [hunk 1]
```diff
@@ -46,13 +46,23 @@ passing, so it is often a good idea to make sure the e2e tests work as well.
...
 `make test` is the entrypoint for running the unit tests that ensures that
-`GOPATH` is set up correctly.  If you have `GOPATH` set up correctly, you can
-also just use `go test` directly.
+`GOPATH` is set up correctly.
 
 ```sh
 cd kubernetes
 make test  # Run all unit tests.
 ```
+If you have `GOPATH` set up correctly, you can
```
Comment (stevekuznetsov) [inline contributors/devel/sig-testing/testing.md]: With a new enough `go`, do you need this?

Comment (fejta) [reply contributors/devel/sig-testing/testing.md]: Doesn't this require the main repo to setup go modules?

Comment (mattjmcnaughton) [reply contributors/devel/sig-testing/testing.md]: I agree that I think that unless go module support is added we still need the GOPATH to be setup.

**Pull request merged**